articles/how-to-prevent-cross-site-scripting-in-node.js/index.md

```diff
@@ -0,0 +1,133 @@
+
+#### Output Encoding
+Output encoding translates untrusted input into a safe model. It takes place by running a JavaScript entity encoding in a Node.js platform server prior to sending untrusted data into the browser. 
+
+#### Avoid Inserting Untrusted Data
+Do not pass untrusted data into executable JavaScript functions such as `eval`. In the Node.js `eval` function, enable code injections. The `eval` function takes a string argument and interprets it as a JavaScript code allowing the attacker to execute it.
+
+The only secure way to use if you insert untrusted data into the JavaScript code is inside a quoted (")data value. That is the proper context for inserting data into a JavaScript code. For example, 
+```js
+<script>Alert (“…encode untrusted data before being executed…”)</script>
+```
+
+#### Validating User Input
+To prevent XSS, developers must [validate user input](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html) and encode the output. Validation of user input is a defense technique used on the server-side to prevent XSS attacks. This means XSS mitigation measures are applied on a node.js platform.
+
+Input validation is performed to make sure only secure data enters an information system. Validation of user input should happen as soon as data is received from the other party. Use the following script to prevent XSS by input validation.
+```js
+var validator = require('validator');
+var escaped_string = validator.escape(string);
+```
+
+Validating input before processing the attacker's request is important because it prevents XSS attacks early. Input validation detects unauthorized user input before the application executes it. Special characters such as (. and \ ) should be filtered to prevent them from being used maliciously.
+
+It is essential to carry out server-side input validation when preventing XSS attacks. Server-side is mainly preferred for input validation because it can easily be bypassed when done via the client-side.
+
+While validating user input with server-side JavaScript (node.js), try the npm module `strip-js`. It does the following:
+- Sanitizes data
+- Do away with script tags
+- Remove attributes that contain JavaScript code, i.e., `onclick` and `onerror`.
+- Remove “href” attributes with JavaScript code.
+
+#### Sanitizing input 
+Properly sanitized input ensures that users can submit only trusted data. It ensures that only expected data is inputted, i.e. (only authorizing input fields to accept parenthesis, numbers, and hyphens).  Sanitizing data in an HTTP request before execution ensures data is validated, escaped, and filtered before it is outputted to the user.
+
+Another way to prevent XSS is preventing cookies from browser JS scripts. This is achieved by setting the cookie `httpOnly` value to `true` in the below code.
```

Comment:
Does this come under sanitizing input section, or is it a different one? It is not clear. Also, explain what happens when the cookie value is set to `True.`